plugins/modules/alert_settings.py

```diff
@@ -0,0 +1,242 @@
+
+DOCUMENTATION = r'''
+---
+module: alert_settings
+version_added: '3.2.0'
+short_description: Manage alert settings on a PowerScale Storage System
+description:
+- Managing alert settings on a PowerScale system includes retrieving details of
+  alert settings and enabling or disabling the alert settings.
+
+extends_documentation_fragment:
+  - dellemc.powerscale.powerscale
+
+author:
+  - Bhavneet Sharma (@Bhavneet-Sharma) <ansible.team@dell.com>
+options:
+  enable_celog_maintenance_mode:
+    description:
+    - Enabling CELOG maintenance mode will start a CELOG maintenance window.
+    - During a CELOG maintenance window, the system will continue to log
+      events, but no alerts will be generated.
+    - You will have the opportunity to review all events that took place
+      during the maintenance window when disabling maintenance mode.
+    - Active event groups will automatically resume generating alerts when
+      the scheduled maintenance period ends.
+    type: bool
+  prune:
+    description:
+    - Removes all maintenance mode history that is greater than set number of days.
+    - Range of I(prune) is 0 to 4294967295.
+    type: int
+
+notes:
+- The I(check_mode) and idempotency is supported.
+'''
+
+EXAMPLES = r'''
+- name: Get alert setiings
+  dellemc.powerscale.alert_settings:
+    onefs_host: "{{ onefs_host }}"
+    port_no: "{{ port_no }}"
+    api_user: "{{ api_user }}"
+    api_password: "{{ api_password }}"
+    verify_ssl: "{{ verify_ssl }}"
+
+- name: Enable CELOG maintenance mode
+  dellemc.powerscale.support_assist:
+    onefs_host: "{{ onefs_host }}"
+    port_no: "{{ port_no }}"
+    api_user: "{{ api_user }}"
+    api_password: "{{ api_password }}"
+    verify_ssl: "{{ verify_ssl }}"
+    enable_celog_maintenance_mode: true
+    prune: 1
+
+- name: Disable CELOG maintenance mode
+  dellemc.powerscale.support_assist:
+    onefs_host: "{{ onefs_host }}"
+    port_no: "{{ port_no }}"
+    api_user: "{{ api_user }}"
+    api_password: "{{ api_password }}"
+    verify_ssl: "{{ verify_ssl }}"
+    enable_celog_maintenance_mode: false
+    prune: 2
+'''
+
+RETURN = r'''
+changed:
+    description: A boolean indicating if the task had to make changes.
+    returned: always
+    type: bool
+    sample: "false"
+alert_settings_details:
+    description: The updated alert settings details.
+    type: dict
+    returned: always
+    contains:
+      history:
+        description: History list of CELOG maintenance mode windows.
+        type: list
+        contains:
+          end:
+            description: End time of CELOG maintenance mode, as a UNIX
+                         timestamp in seconds.
+                         Value 0 indicates that maintenance
+                         mode is still enabled.
+            type: int
+          start:
+            description: Start time of CELOG maintenance mode, as a UNIX
+                         timestamp in seconds.
+            type: int
+          network_pools:
```

Comment:
Do we have network pools also in response?

Reply:
Done.